src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -502,6 +502,23 @@ private function isInteractiveInput($inputStream): bool
      */
     private function readInput($inputStream, Question $question): string|false
     {
+        if (null !== $question->getTimeoutSeconds()) {
+            $streamType = stream_get_meta_data($inputStream)['stream_type'] ?? null;
+
+            if ('MEMORY' === $streamType) {
+                throw new RuntimeException('Using question timeout is incompatible with memory streams.');
```

Comment:
Maybe, we could allow timeout only when `$this->isInteractiveInput($inputStream)` returns `true`...